relayer/relayer.go

```diff
@@ -161,34 +159,39 @@ func NewRelayer(
 }
 
 // Helper to process missed blocks. Chooses as the starting block the maximum of the latest processed block and the configured start block height.
-// startBlockHeight cannot be nil
+// If the DB does not contain a value, startBlockHeight cannot be nil
 func (r *Relayer) processMissedBlocks(
 	sub vms.Subscriber,
 	startBlockHeight *big.Int,
 ) error {
-	if startBlockHeight == nil {
-		return fmt.Errorf("startBlockHeight cannot be nil")
-	}
 	// Attempt to get the latest processed block height from the database.
 	// Note that the retrieved latest processed block may have already been partially (or fully) processed by the relayer on a previous run. When
 	// processing a warp message in real time, which is when we update the latest processed block in the database, we have no way of knowing
 	// if that is the last warp message in the block
-	latestProcessedBlockData, err := r.db.Get(r.sourceBlockchainID, []byte(database.LatestProcessedBlockKey))
 
 	// First, determine the height to process from. There are two cases:
 	// 1) The database contains the latest processed block data for the chain
 	//    - In this case, we process from the maximum of the latest processed block and the configured start block height to the latest block
 	// 2) The database has been configured for the chain, but does not contain the latest processed block data
 	//    - In this case, we process from the start block height to the latest block
-	height := startBlockHeight
-	if err == nil {
-		// Use the max of the latest processed block and the start block height
+	var height *big.Int
+	latestProcessedBlockData, err := r.db.Get(r.sourceBlockchainID, []byte(database.LatestProcessedBlockKey))
+	if errors.Is(err, database.ErrChainNotFound) || errors.Is(err, database.ErrKeyNotFound) {
+		// The database does not contain the latest processed block data for the chain, so use the configured StartBlockHeight instead
+		if startBlockHeight == nil {
+			r.logger.Warn("database does not contain latest processed block data and startBlockHeight is nil. Please provide a StartBlockHeight in the configuration.")
+			return errors.New("database does not contain latest processed block data and startBlockHeight is nil.")
+		}
+		height = startBlockHeight
+	} else if err == nil {
```

Comment:
nit: if we move the else clause on line 209 up here to be `else if err != nil`, and then we can unindent the happy path.

Reply:
Nice spot, made this change